### ISSUE TITLE
About using hypervolume

### ISSUE THREAD

**Issue by henryyuri** [MOEAFramework/MOEAFramework#77]
Hello David,

When I executed the program with external hypervolume calculators (either wfg2.exe or hoy.exe) by following the steps on https://github.com/MOEAFramework/Hypervolume and I got errors:

Exception in thread "main" java.lang.NullPointerException
	at org.moeaframework.core.indicator.Hypervolume.invokeNativeProcess(Hypervolume.java:473)
	at org.moeaframework.core.indicator.Hypervolume.invokeNativeHypervolume(Hypervolume.java:436)
	at org.moeaframework.core.indicator.Hypervolume.evaluate(Hypervolume.java:338)
	at org.moeaframework.core.indicator.Hypervolume.evaluate(Hypervolume.java:298)
	at org.moeaframework.Analyzer.getAnalysis(Analyzer.java:853)
	at org.moeaframework.Analyzer.printAnalysis(Analyzer.java:954)
	at org.moeaframework.Analyzer.printAnalysis(Analyzer.java:682)
	at Example1.main(Example1.java:146)
C:\Users\henryyuri\AppData\Local\NetBeans\Cache\8.1\executor-snippets\run.xml:53: Java returned: 1

The code is as follows:

		Executor executor = new Executor()
		.withProblemClass(DTLZ1.class,10)
		.withProperty("populationSize", 200)
		.withMaxEvaluations(140000);
		Analyzer analyzer = new Analyzer()
		.withProblemClass(DTLZ1.class,10)
                 .includeHypervolume()
		.showIndividualValues();
                for (String algorithm : algorithms) {
			analyzer.addAll(algorithm, 
					executor.withAlgorithm(algorithm).runSeeds(30));
		}
                
		analyzer.printAnalysis();

I know the hoy.exe requires 4 arguments......but how can I plug the arguments in the above case? Many thanks.

**Comment by dhadka** [comment]
Hi,

Please make sure your `global.properties` file contains the following two lines exactly as shown below:

```
org.moeaframework.core.indicator.hypervolume = ./wfg2 {2}
org.moeaframework.core.indicator.hypervolume_inverted = true
```

In particular, make sure it has the `{2}`.  Without the `{2}`, no input file is provided to the wfg2 executable.

If you're still having issues, please attach your `global.properties` file and also copy the entire output (the exception plus any other output) to this ticket and I can investigate further.

Thanks!
Dave

**Comment by henryyuri** [comment]
Thanks a lot. 

I tried either
org.moeaframework.core.indicator.hypervolume = ./wfg2 {2}
org.moeaframework.core.indicator.hypervolume = ./wfg2.exe {2}

However, problems exist. Following are the output:

run:
Exception in thread "main" java.lang.NullPointerException
	at org.moeaframework.core.indicator.Hypervolume.invokeNativeProcess(Hypervolume.java:473)
	at org.moeaframework.core.indicator.Hypervolume.invokeNativeHypervolume(Hypervolume.java:436)
	at org.moeaframework.core.indicator.Hypervolume.evaluate(Hypervolume.java:338)
	at org.moeaframework.core.indicator.Hypervolume.evaluate(Hypervolume.java:298)
	at org.moeaframework.Analyzer.getAnalysis(Analyzer.java:853)
	at org.moeaframework.Analyzer.printAnalysis(Analyzer.java:954)
	at org.moeaframework.Analyzer.printAnalysis(Analyzer.java:682)
	at Example11.main(Example11.java:136)
C:\Users\user\AppData\Local\NetBeans\Cache\8.1\executor-snippets\run.xml:53: Java returned: 1
BUILD FAILED (total time: 1 second)

[global.properties.txt](https://github.com/MOEAFramework/MOEAFramework/files/530305/global.properties.txt)

**Comment by henryyuri** [comment]
Hello David,

I compiled the hoy using the MinGW compiler (using the command: g++ -o hoy hoy.cpp) and the resulted hoy.exe is compatible with the MOEA Framework. 

However, when I tried to compile the wfg codes and fails. 
When I type mingw32-make in command prompt,
gcc -std=c99 -Wall -O3 -DSIMPLE -Dopt=0 -o wfg0 wfg.c read.o avl.o
read.o: file not recognized: File format not recognized
collect2.exe: error: ld returned 1 exit status
makefile:6: recipe for target 'compile' failed
mingw32-make: *** [compile] Error 1

When I type gcc wfg.c,
wfg.c:38:26: fatal error: sys/resource.h: No such file or directory
compilation terminated.

On the other hand, I compiled the wfg codes by using cygwin, 4 new files are created: wfg0.exe to wfg3.exe. However, the wfg2.exe is not compatible with the MOEA Framework and the error occurs as I mentioned at first in this issue.

Maybe I will use hoy to calculate the hypervolume value of the obtained Pareto fronts. Many thanks for your help.
